website/pages/en/index.js

```diff
@@ -29,32 +29,36 @@ const pre = "```";
 
 // fake, static, responsive refmt, lol. See reason.css homeCodeSnippet logic
 const codeExampleSmallScreen = `${pre}reason
-type schoolPerson =
-  | Teacher
-  | Director
-  | Student(string);
-
-let greeting = person =>
-  switch (person) {
-  | Teacher => "Hey Professor!"
-  | Director => "Hello Director."
-  | Student("Richard") =>
-    "Still here Ricky?"
-  | Student(anyOtherName) =>
-    "Hey, " ++ anyOtherName ++ "."
+/* Paste into online playground and try it out: */
+type track = int;
+type playState =
+  | Playing(track)
+  | Paused(track)
+  | NotPlaying;
+
+/* ~newTrack is a labeled optional argument! */
+let togglePlay = (~newTrack=1, playState) =>
+  switch (playState) {
+  | Playing(track) => Paused(track)
+  | Paused(track) => Playing(track)
+  | NotPlaying => Playing(newTrack)
   };
+let nextPlayState = togglePlay(NotPlaying);
```

Comment:
Do you think that expressing the same behavior with a state machine would demonstrate the power of pattern matching, or would that be too large of a snippet for the front page?

[PlayerMachine.re](https://sketch.sh/s/XcxUxtZ7ofU22ppuvM48x3/)
```reason
type track = int;
type status = 
  | Playing(track)
  | Paused(track)
  | Stopped;
type event =
  | Play(track)
  | Resume
  | Pause
  | Stop;

let send = (event, status) => 
  switch(status, event) {
    | (_, Play(track)) 
    | (Playing(track), Resume)
    | (Paused(track), Resume) => Playing(track)
    | (Paused(track), Stop) => Stopped
    | (Paused(track), Pause)
    | (Playing(track), Pause) => Paused(track)
    | (Playing(_), Stop)
    | (Stopped, Resume)
    | (Stopped, (Stop|Pause)) => Stopped
  };
  
 Stopped
  |> send(Play(0)) /* => Playing(0) */
  |> send(Pause)   /* => Paused(0) */
  |> send(Resume)  /* => Playing(0) */
  |> send(Stop);   /* => Stopped */
```

Reply:
Good question; I deliberately kept it very minimal to fit above the fold and make it reasonably possible to grok in a single pass.

Reply:
Makes sense. I'll keep this example in mind for a more advanced section. I think it wise to position Reason as an ideal language for expressing state machines given their recent rise in popularity in the frontend.